pkg/apis/operator/v1alpha1/tektonconfig_validation.go

```diff
@@ -62,10 +62,12 @@ func (p Prune) validate() *apis.FieldError {
 		errs = errs.Also(apis.ErrMissingField("spec.pruner.resources"))
 	}
 
-	if p.Keep == nil {
-		errs = errs.Also(apis.ErrMissingField("spec.pruner.keep"))
-	} else if *p.Keep == 0 {
+	if p.Keep == nil && p.KeepSince == nil {
+		errs = errs.Also(apis.ErrMissingOneOf("spec.pruner.keep", "spec.pruner.keep-since"))
```

Comment:
we also need to check if both are specified together. That should return an error, as we can accept only one of `keep` of `keepSince`

Reply:
Yeh, it makes sense, Thank You

Reply:
This check is still needed isn't it ? if we specify both should it fail or ?

Reply:
if both are specified, it will return a validation error. 
or we can split this into
```
p.strategy = keep/keepSince
p.keep
p.keepSince
```

either one of p.keep and p.keepSince, could be used depending on p.strategy.
we are following a similar approach in overides for the same using namespace annotations.

Reply:
In the code, I don't see that check — that it will fail if both are present in `Tektonconfig` CR.

Reply:
On the "strategy", what if, someday, we want to allow both ? we would add a new strategy ?

Reply:
>In the code, I don't see that check — that it will fail if both are present in Tektonconfig CR.
I am addressing the review comments and it will be taken care in the updated pr.

Reply:
> On the "strategy", what if, someday, we want to allow both? we would add a new strategy ?

Let's not introduce strategy at the tektonconfig level.